File: shardis-web/src/main/java/com/shardis/web/controllers/MainController.java
package com.shardis.web.controllers;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.core.io.Resource;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/")
public class MainController {

    private final Resource indexHtml;

    public MainController(@Value("classpath:/static/index.html") Resource indexHtml) {
        this.indexHtml = indexHtml;
    }

    @GetMapping("/")
    Resource index() {
        return this.indexHtml;
    }
}

File: shardis-web/src/main/kotlin/com/shardis/web/controllers/MainController.kt
package com.shardis.web.controllers

import org.springframework.beans.factory.annotation.Value
import org.springframework.core.io.Resource
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/")
class MainController(@param:Value("classpath:/static/index.html") private val indexHtml: Resource) {

    @GetMapping("/")
    fun index() = indexHtml
}
